src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -407,7 +403,7 @@ private bool AcquireClientCredentials(ref byte[] thumbPrint)
                 }
                 else
                 {
-                    if (ClientCertificates.Count == 0)
+                    if (_clientCertificates == null || _clientCertificates.Count == 0)
```

Comment:
You are right (never attempt a review on a mobile phone). So I am looking at the code on my computer now, this isn't the method I am really worried about. It's more the next one AcquireClientCredentials that is many pages long, and nested a number of levels deep. It looks a bit "openssl". So I have no problem with the changes in general, just wondering if while this is being looked at if this could be refactored in some way.  Anyway the changes you have made look good and are an improvement.